osfoffline/views/preferences.py

```diff
@@ -36,12 +35,39 @@ def __init__(self):
 
         self.preferences_window.changeFolderButton_2.clicked.connect(self.update_sync_nodes)
         self.tree_items = []
+        self.checked_items = []
         self.setup_slots()
 
+    def get_guid_list(self):
+        guid_list = []
+        for tree_item in self.tree_items:
+            for name, id in [(node.name, node.id) for node in self.remote_top_level_nodes]:
+                if name == tree_item.text(self.PROJECT_NAME_COLUMN):
```

Comment:
This function just gets the node associated with a project name. A note, but also a possible bug:

One way to boost performance would be to create a dictionary representation of `remote_top_level_names` before looping over tree items (to save iteration every time):
`node_lookup = {node.name : node.id for node in self.remote_top_level_nodes}`

BUT: what does this function do if two projects (with different node IDs) have the same name?